selffuzz/src/test/java/com/code_intelligence/selffuzz/driver/FuzzedDataProviderImplFuzzTest.java

```diff
@@ -0,0 +1,433 @@
+import com.code_intelligence.selffuzz.jazzer.driver.FuzzedDataProviderImpl;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+public class FuzzedDataProviderImplFuzzTest {
+  @FuzzTest
+  void fuzzedDataProviderTest(byte @NotNull[] buf) {
+    try (FuzzedDataProviderImpl data = FuzzedDataProviderImpl.withJavaData(buf)) {
+      ArrayList<Consumer<FuzzedDataProvider>> actionList = getActionList();
+
+      while (data.remainingBytes() > 0) {
+        int actionIndex = data.consumeInt(0, actionList.size() - 1);
+        Consumer<FuzzedDataProvider> action = actionList.get(actionIndex);
+        action.accept(data);
+      }
+    }
+  }
+
+  ArrayList<Consumer<FuzzedDataProvider>> getActionList() {
+    ArrayList<Consumer<FuzzedDataProvider>> actionList = new ArrayList<>();
+    actionList.add(this::testBoolean);
+    actionList.add(this::testBooleans);
+    actionList.add(this::testByte);
+    actionList.add(this::testByteMinMax);
+    actionList.add(this::testBytes);
+    actionList.add(this::testRemainingAsBytes);
+    actionList.add(this::testShort);
+    actionList.add(this::testShortMinMax);
+    actionList.add(this::testShorts);
+    actionList.add(this::testInt);
+    actionList.add(this::testIntMinMax);
+    actionList.add(this::testInts);
+    actionList.add(this::testLong);
+    actionList.add(this::testLongMinMax);
+    actionList.add(this::testLongs);
+    actionList.add(this::testFloat);
+    actionList.add(this::testRegularFloat);
+    actionList.add(this::testRegularFloatMinMax);
+    actionList.add(this::testProbabilityFloat);
+    actionList.add(this::testDouble);
+    actionList.add(this::testRegularDouble);
+    actionList.add(this::testRegularDoubleMinMax);
+    actionList.add(this::testProbabilityDouble);
+    actionList.add(this::testChar);
+    actionList.add(this::testConsumeCharMinMax);
+    actionList.add(this::testCharNoSurrogates);
+    actionList.add(this::testString);
+    actionList.add(this::testRemainingAsString);
+    actionList.add(this::testAsciiString);
+    actionList.add(this::testRemainingAsAsciiString);
+    actionList.add(this::testPickValueCollection);
+    actionList.add(this::testPickValueArray);
+    actionList.add(this::testPickValueBoolean);
+    actionList.add(this::testPickValueByte);
+    actionList.add(this::testPickValueShort);
+    actionList.add(this::testPickValueInt);
+    actionList.add(this::testPickValueLong);
+    actionList.add(this::testPickValueDouble);
+    actionList.add(this::testPickValueFloat);
+    actionList.add(this::testPickValueChar);
+    actionList.add(this::testPickValuesCollection);
+    actionList.add(this::testPickValuesArray);
+    return actionList;
+  }
+
+  void testBoolean(FuzzedDataProvider data) {
+    data.consumeBoolean();
+  }
+
+  void testBooleans(FuzzedDataProvider data) {
+    try {
+      data.consumeBooleans(data.consumeInt());
+    } catch (IllegalArgumentException e) {
+      // catch negative length
```

Comment:
This can hide actual bugs. Could we instead just return early if `data.consumeInt() < 0`?

The same applies to the other cases of `catch (IllegalArgumentException e)` below.